packages/clay-label/src/ClayLabel.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+
+type DisplayType = 'secondary' | 'info' | 'warning' | 'danger' | 'success';
+
+interface Props extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
+	closeButtonProps?: React.HTMLAttributes<HTMLButtonElement>;
+	displayType?: DisplayType;
+	large?: boolean;
+}
+
+const ClayLabel: React.FunctionComponent<Props> = ({
```

Comment:
I'm finding this component a little tricky and maybe we do not need to call `React.createElement` directly. We can simplify a little:

Just a suggestion, what do you think? 🙂
```javascript
const ClayLabelContent = ({
	children,
	closeButtonProps,
}) => (
	<>
		<span className="label-item label-item-expand">{children}</span>
		{closeButtonProps && (
			<span className="label-item label-item-after">
				<button
					{...closeButtonProps}
					className="close"
					type="button"
				>
					{closeButtonProps.children}
				</button>
			</span>
		)}
	</>
);

const ClayLabel = ({
	children,
	className,
	closeButtonProps,
	displayType = 'secondary',
	href,
	large = false,
	...otherProps
}) => {
	const classNames = cN('label', className, {
		'label-dismissible': closeButtonProps,
		'label-lg': large,
		[`label-${displayType}`]: displayType,
	});

	if (href) {
		return (
			<a
				href={href}
                                 className={classNames}
				{...otherProps}
			>
				<ClayLabelContent closeButtonProps={closeButtonProps}>
					{children}
				</ClayLabelContent>
			</a>
		);
	}

	return (
		<span {...otherProps} className={classNames}>
			<ClayLabelContent closeButtonProps={closeButtonProps}>
				{children}
			</ClayLabelContent>
		</span>
	);
}
```

Reply:
Hey Matu, what exactly do you find confusing? Is it particularly the call to `React.createElement`? I was trying to avoid multiple returns and also avoiding duplicating the markup for the `content`.

Reply:
Hey Bryce, yeah I understand what you wanted to do, the big reason is that I do not really like having to call `React.createElement` directly, I'd rather let the babel presets to react transform to `React.createElement`, in this case, and I find it a bit strange that we have a condition that removes the href from the above object (it seems a bit hack), I think we have other ways of doing this without having to compromise a little beauty of our code in the development environment.

I think we do not need to focus as much on eliminating as many rows (we can then improve our build steps for distribution) by just getting in the middle so we do not remove some beauty from the code.

> Just making sure that we can have other ways of doing this.

Reply:
Good point. I actually was able to simplify it by simply using `const TagName = href ? 'a' : 'span';` I think this way makes most sense to me.

Reply:
Oh it looks much better now, good move with `TagName`.